phpunit/class-gutenberg-utils-test.php

```diff
@@ -121,4 +121,11 @@ public function test_invalid_parameters_set() {
 			array( 'a' => 2 )
 		);
 	}
+
+	public function test_gutenberg_experimental_to_kebab_case() {
+		$this->assertEquals( 'white', gutenberg_experimental_to_kebab_case( 'white' ) );
+		$this->assertEquals( 'white-2-white', gutenberg_experimental_to_kebab_case( 'white2white' ) );
+		$this->assertEquals( 'white-2nd', gutenberg_experimental_to_kebab_case( 'white2nd' ) );
```

Comment:
I don't think we need to be so ambitious in imitating `kebabCase` as to be aware of numerals. Lodash uses a [lexicon of words](https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/.internal/unicodeWords.js) that it recognizes and treats differently, but that adds a significant amount of complexity.

I think we can start smaller, and maybe recommend somewhere in the docs against using numerals. More realistic cases -- such as `2xl` or `h2` -- will be treated equally by `kebabCase` and our PHP-side implementation.

```suggestion
		$this->assertEquals( 'white-2-xl', gutenberg_experimental_to_kebab_case( 'white2xl' ) );
```

Reply:
Oh wow, I noticed you just added the lexicon this very moment 😅

Reply:
Yeah. I've also added a test for 2xl below.

There're a couple of things I'd like to check:

- what use case is covered by `rsEmoji`
- why `rsAstralRange` needs to be different than lodash
- whether there're more things we can remove safely
- how the changes in the latest published lodash version differ from the one we use ([4.17](https://github.com/lodash/lodash/blob/4.17/dist/lodash.js#L278)) and whether that will affect us (see `rsDigit`)

Other than that it's almost ready.

PS: while I've linked to [lodash-php](https://github.com/lodash-php/lodash-php/blob/master/src/internal/unicodeWords.php) for reference, this code follows lodash's kebabCase which is different than the PHP version (it does a few things differently, such as the numerals regexp).

Reply:
I've pushed a version that removed all the non-essential regexp we need to support and added comments as necessary.